lib/spack/spack/test/spec_yaml.py

```diff
@@ -350,7 +346,7 @@ def test_save_dependency_spec_jsons_subset(tmpdir, config):
         spec_a.concretize()
         b_spec = spec_a['b']
         c_spec = spec_a['c']
-        spec_a_json = spec_a.to_json(hash=ht.build_hash)
+        spec_a_json = spec_a.to_json(hash=ht.dag_hash)
```

Comment:
Can we:
```suggestion
        spec_a_json = spec_a.to_json()
```
since `ht.dag_hash` is the default?